dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java

```diff
@@ -718,7 +730,7 @@ private void writeDecimal(FieldDef fd, BigDecimal decimalValue)
 
         // Strip leading zeros. We only want sig digits
         int numLeadingZeros = 0;
-        while (decimalStr.charAt(numLeadingZeros) == '0')
+        while (decimalStr.length()>numLeadingZeros && decimalStr.charAt(numLeadingZeros) == '0')
```

Comment:
do we need to ensure decimalValue != null early on?

Reply:
This method's called only from writeField, and the line before it's callled makes sure it's not null:
BigDecimal value = fieldValue==null?BigDecimal.valueOf(0):(BigDecimal) fieldValue;
writeDecimal(fd, value);